### PR TITLE
TYP: check_untyped_defs pandas.core.nanops

### PR DESCRIPTION
**Pull request by simonjayhawkins**
pandas\core\nanops.py:565: error: Item "bool" of "Union[bool, Any]" has no attribute "any"

### CODE REVIEW

**Review by WillAyd — COMMENTED** [review]
minor question otherwise looks good

File: pandas/core/nanops.py [hunk 1]
```diff
@@ -557,7 +562,8 @@ def nanmean(values, axis=None, skipna=True, mask=None):
     count = _get_counts(values.shape, mask, axis, dtype=dtype_count)
     the_sum = _ensure_numeric(values.sum(axis, dtype=dtype_sum))
 
-    if axis is not None and getattr(the_sum, "ndim", False):
+    if not is_scalar(count):
```
[the remark this answers is not in the document]
Comment (simonjayhawkins) [reply pandas/core/nanops.py:565]: IIUC, if count is a scalar, ct_mask = count == 0 would evaluate to a bool and then ct_mask.any() would raise an AttributeError.

pandas\core\nanops.py:571: error: Item "bool" of "Union[bool, Any]" has no attribute "any"

File: pandas/core/nanops.py [hunk 2]
```diff
@@ -514,7 +514,12 @@ def nansum(
 
 @disallow(PeriodDtype)
 @bottleneck_switch()
-def nanmean(values, axis=None, skipna=True, mask=None):
+def nanmean(
+    values: np.ndarray,
+    axis: Optional[int] = None,
+    skipna: bool = True,
+    mask: np.ndarray = None,
```
[the remark this answers is not in the document]
Comment (simonjayhawkins) [reply pandas/core/nanops.py:521]: yep. good catch. np.ndarray evaluates to Any, so None is compatible. will update.

File: pandas/core/nanops.py [hunk 3]
```diff
@@ -130,7 +130,7 @@ def f(
 
             return result
 
-        return f
+        return cast(F, f)
```
[the remark this answers is not in the document]
Comment (simonjayhawkins) [reply pandas/core/nanops.py:133]: This is how we have been typing all our decorators to preserve the signature. Its recommended best practice. https://mypy.readthedocs.io/en/stable/generics.html#declaring-decorators

if we don't cast, we get a mypy error `pandas\core\nanops.py:133: error: Incompatible return value type (got "Callable[..., Any]", expected "F")`

File: pandas/core/nanops.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (simonjayhawkins) [reply pandas/core/nanops.py:565]: reverted.

**Comment by simonjayhawkins** [comment]
@WillAyd @jorisvandenbossche updated + green

**Comment by WillAyd** [comment]
Lgtm @jorisvandenbossche merge if nothing outstanding

**Comment by jreback** [comment]
thanks @simonjayhawkins

**Pull request merged**